File: Iteration1/SourceCode/src/main/java/CourseSection.java
import java.util.ArrayList;

public class CourseSection {
    private int quota;
    private Lecturer sectionLecturer;
    private Course course;
    private int sectionNumber;
    private String courseHours;
    private ArrayList<CourseSectionRegister> registeredStudents= new ArrayList<>();
    CourseSection(Course course,int sectionNumber,int quota,Lecturer sectionLecturer,String courseHours){
        this.course = course;
        this.sectionNumber=sectionNumber;
        this.quota=quota;
        this.sectionLecturer=sectionLecturer;
        this.courseHours=courseHours;
    }

    public void setCourse(Course course) {
        this.course = course;
    }

    public int getSectionNumber() {
        return sectionNumber;
    }

    public void setSectionNumber(int sectionNumber) {
        this.sectionNumber = sectionNumber;
    }

    public String getCourseHours() {
        return courseHours;
    }

    public void setCourseHours(String courseHours) {
        this.courseHours = courseHours;
    }


    public int getQuota() {
        return quota;
    }

    public void setQuota(int quota) {
        this.quota = quota;
    }

    public Lecturer getSectionLecturer() {
        return sectionLecturer;
    }

    public void setSectionLecturer(Lecturer sectionLecturer) {
        this.sectionLecturer = sectionLecturer;
    }

    public Course getCourse() {
        return course;
    }
    public void requestToRegister(Student student){
        CourseSectionRegister courseSectionRegister = new CourseSectionRegister(this,"register",student);

    }
    public boolean isQuotaAvailable(){
        if(getRegisteredStudents().size()<getQuota()){
            return true;
        }
        return false;
    }
    public void addToRegistrationList(CourseSectionRegister register){
        ArrayList <CourseSectionRegister> newList = this.getRegisteredStudents();
        newList.add(register);
        this.setRegisteredStudents(newList);
    }

    public ArrayList<CourseSectionRegister> getRegisteredStudents() {
        return registeredStudents;
    }

    public void setRegisteredStudents(ArrayList<CourseSectionRegister> registeredStudents) {
        this.registeredStudents = registeredStudents;
    }
}

File: Iteration3/SourceCode/CourseSection.py
from CourseSectionRegister import CourseSectionRegister


class CourseSection(object):
    def __init__(self, course, sectionNumber, quota, sectionLecturer, courseHours):
        self.__registeredStudents = []

        self.__course = course
        self.__sectionNumber=sectionNumber
        self.__quota=quota
        self.__sectionLecturer=sectionLecturer
        self.__courseHours=courseHours

    def getCourse(self):
        return self.__course
    
    def setCourse(self, course):
        self.__course = course

    def getSectionNumber(self):
        return self.__sectionNumber

    def setSectionNumber(self, sectionNumber):
        self.__sectionNumber = sectionNumber

    def getCourseHours(self):
        return self.__courseHours

    def setCourseHours(self, courseHours):
        self.__courseHours = courseHours


    def getQuota(self):
        return self.__quota

    def setQuota(self, quota):
        self.__quota = quota

    def getSectionLecturer(self):
        return self.__sectionLecturer

    def setSectionLecturer(self, sectionLecturer):
        self.__sectionLecturer = sectionLecturer

    def getCourse(self):
        return self.__course



    def requestToRegister(self, student):
        courseSectionRegister = CourseSectionRegister(self,"register",student)
    def isQuotaAvailable(self):
        if len(self.getRegisteredStudents())<self.getQuota():
            return True
        return False
    def addToRegistrationList(self, register):
        newList = self.getRegisteredStudents()
        newList.append(register)
        self.setRegisteredStudents(newList)

    def getRegisteredStudents(self):
        return self.__registeredStudents

    def setRegisteredStudents(self, registeredStudents):
        self.__registeredStudents = registeredStudents